mono/mini/method-to-ir.c

```diff
@@ -6421,12 +6421,17 @@ emit_setret (MonoCompile *cfg, MonoInst *val)
 		} else {
 			EMIT_NEW_RETLOADA (cfg, ret_addr);
 
+#ifdef MONO_ARCH_SIMD_INTRINSICS
```

Comment:
This change should not cause the code to behave differently.

Simplified from mini.h: 

```
#ifdef MONO_ARCH_SIMD_INTRINSICS
#define MONO_CLASS_IS_SIMD(cfg, klass) (((cfg)->opt & MONO_OPT_SIMD) && m_class_is_simd_type (klass))
#else
#define MONO_CLASS_IS_SIMD(cfg, klass) (0)
#endif
```

Meaning if `MONO_ARCH_SIMD_INTRINSICS` is not defined, the existing code will always execute the else branch and emit the same code as proposed by this change.

Additionally, 8dd6baee6e7f change to method-to-ir.c should use MONO_CLASS_IS_SIMD instead of adding a `#ifdef MONO_ARCH_SIMD_INTRINSICS`